### PR TITLE
Helper functions for macros

### PR DESCRIPTION
**Pull request by daroczig**
I've found these functions created in my internal package really useful when auto-generating docs from R, so I thought others might find it useful as well -- let me know what you think.

With the added (internal) `conf_macro_generator` fn it's really easy to add support for further macros, and I'm happy to add more in the future as some further cases came up, just let me know if this is something you might be interested.

### CODE REVIEW

File: R/macros.R [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    inline = '`',
+    block = '\n```{=html}\n')
+
+  macro <- glue('{macro}<ac:structured-macro ac:name="{name}">')
+
+  if (!missing(parameters)) {
+    for (parameter in names(parameters)) {
+      macro <- paste0(
+        macro,
+        glue('<ac:parameter ac:name="{parameter}">'),
```
Comment (yutannihilation) [inline R/macros.R:23]:
```suggestion
        glue::glue('<ac:parameter ac:name="{parameter}">'),
```

File: R/macros.R [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        parameters[[parameter]],
+        '</ac:parameter>')
+    }
+  }
+
+  if (!missing(body)) {
```
Comment (yutannihilation) [inline R/macros.R:29]: This is just a matter of preference, but I prefer setting `NULL` as the default of optional arguments and check `is.null()`. Could you follow this way?

c.f. https://design.tidyverse.org/def-required.html

File: R/macros.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    macro <- paste0(
+      macro,
+      '<confl-ac-rich-text-body>',
```
Comment (yutannihilation) [inline R/macros.R:32]: I think you can use `<ac:...>` here as well. Are there any reason to use `confl-ac`? (Just curious)

File: R/macros.R [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+
+#' Generate Confluence macro referencing a Jira ticket
+#' @param key Jira ticket id, eg CONFLR-XXXX
+#' @return HTML as string
+#' @export
+#' @examples
+#' confl_macro_jira('CONFLR-42')
+confl_macro_jira <- function(key) {
+  conf_macro_generator(type = 'inline', name = 'jira', parameters = list(key = id))
```
Comment (yutannihilation) [inline R/macros.R:66]:
```suggestion
  conf_macro_generator(type = 'inline', name = 'jira', parameters = list(key = key))
```

File: R/macros.R [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#' General Confluence macro builder for internal use
```
Comment (yutannihilation) [inline R/macros.R:1]: (We need a license header here, but I'll do it later)

File: R/macros.R (hunk 3 above)
Comment (daroczig) [reply R/macros.R]: Good catch! Was left there from an earlier version, indeed should standardize over to `ac:`

File: R/macros.R (hunk 1 above)
Comment (daroczig) [reply R/macros.R:23]: Why the need to specify the `glue` package here in such an explicit way?

I have not seen a local `glue` function defined in the `conflr` package, and this PR imports `glue` into the `conflr` pkg namespace from `glue` -- so it should just work fine.

No hard preference ... but I have not really used `::` when it was not really required in other packages, so I wonder why you prefer that.

File: R/macros.R (hunk 4 above)
Comment (daroczig) [reply R/macros.R]: :facepalm:

File: R/macros.R (hunk 1 above)
Comment (yutannihilation) [reply R/macros.R:23]: Ah, importing glue is fine. There's no opinionated reason that I didn't import it. But..., 

> this PR imports `glue` into the `conflr` pkg namespace from `glue`

maybe you forgot to include the related changes in this PR?

Comment (daroczig) [reply R/macros.R:23]: > maybe you forgot to include the related changes in this PR?

definitely :man_facepalming: 

just pushed -- but I can revert to `::` as well (or update all the other `glue::glue` cases to drop the explicit pkg reference), just let me know what you prefer with this

**Comment by daroczig** [comment]
Updated the `NEWS` file, but just realized you already added a comment on the `minor_edit` -- I can fix that later today and resolve the git conflict, by removing my version and keeping yours, sorry for the trouble.

Let me know if I should also add a `ctb` reference in the `DESCRIPTION` file for these minor contributions (though I don't want to push this, I'm not sure what policy you follow on that): https://journal.r-project.org/archive/2012-1/RJournal_2012-1_Hornik~et~al.pdf

**Comment by yutannihilation** [comment]
Thanks! For `ctb` reference, I don't have a concrete policy, but I vaguely follow the tidyverse-like way; use `NEWS.md` to refer to the contributors and list only the core developers' names on `DESCRIPTION`. So, could you add `, @daroczig` to the news item?

**Comment by yutannihilation** [comment]
Perfect! Thanks so much!

**Pull request merged**